src/planet-a/IcoSphereTree.h

```diff
@@ -67,8 +81,70 @@ static constexpr int gc_icosahedronVertCount = 12;
 static constexpr std::uint8_t gc_triangleMaskSubdivided = 0b0001;
 //static constexpr std::uint8_t gc_triangleMaskChunked    = 0b0010;
 
-// An icosahedron with subdividable faces
-// it starts with 20 triangles, and each face can be subdivided into 4 more
+/**
+ * A simple 1D translation and scale
+ */
+struct TriangleSideTransform
+{
+    float m_translation{0.0f};
+    float m_scale{1.0f};
+};
+
+class IcoSphereTreeObserver
+{
+public:
+    using Handle_T = std::list<std::weak_ptr<IcoSphereTreeObserver>>::iterator;
+
+    virtual void on_ico_triangles_added(std::vector<trindex_t> const&) = 0;
+    virtual void on_ico_triangles_removed(std::vector<trindex_t> const&) = 0;
+    virtual void on_ico_vertex_removed(std::vector<vrindex_t> const&) = 0;
+};
+
+
+/**
+ * A triangle on the IcoSphereTree
+ */
+struct SubTriangle
+{
+    trindex_t m_parent;
+    uint8_t m_siblingIndex; // 0:Top 1:Left 2:Right 3:Center
```

Comment:
Looks like a good place to use an enum.

enum class Sibling : std::uint8_t
{
    Top = 0,
    Left = 1,
    Right = 2,
    Center = 3
};

Reply:
There are similar patterns all over the code, where a uint8_t is used to denote a side of a triangle, or siblings. Arithmetic often is performed on these, like on line 560:
```cpp
out.m_translation = out.m_translation * 0.5f
                              + 0.5f * (side == (pTri->m_siblingIndex + 1) % 3);
```
All of these add and modulus operations can be wrapped into some kind of 'cycle' function.

Reply:
Sure, but it should still be an enum :-P

Reply:
A non-class enum? Enum classes are great but when the primary purpose of the enum is to use it as a named integral constant that is involved in arithmetic, it's sort of irritating to insert a verbose C++ cast every time it's used. I can see making it a C-enum though.